### DISCUSSION TITLE
Option to toggle between dark / light mode in docs

### DISCUSSION THREAD

**Discussion by jagira** [rails/rails#51402] [Docs]
I have system wide dark mode enabled. However, I prefer reading web pages and books in light mode. There might be others who have enabled system wide light mode while they prefer reading in dark mode.

Can we have a switch to toggle between dark and light modes? Currently it picks the system setting by default.

**Comment by AmandaPerino** [comment]
Hi @jagira! I prefer that too! :)
It's on the wishlist for a future update.
Thanks for your suggestion.

**Reply by Labern** [reply]
Hi, are there any updates on this? I'd be happy to contribute a working version via a pull request if there is a guideline to do so (e.g. where such a toggle would go, or whether a different solution is in the works at the level of media query). 

I find I have to create a manual solution using DevTools every time I use the guides, and no Chrome extensions are doing the trick, so I'd much rather come up with a solution that others can use too. No worries if not!

**Reply by jeromedalbert** [reply]
Dark Reader on Chrome works pretty well for me if I set the mode to Dynamic

**Reply by Labern** [reply]
Thanks for the idea — couldn't get that to work for me, even after quite a lot of time with it. I really think a simple switch would work for everyone. I was thinking the Rails Guides could implement a basic two-way switch. I'm guessing, and agree, that the priority of the site's `header`, `#page-header`, would be to push such a switch to the far side of the "Contribute". I mocked this up:

![After](https://github.com/user-attachments/assets/00bcbaee-ca80-4922-9aae-5fb2d0161681)

Where "switch" is whatever the final switch would be. 

The only code required (along with the JS implementation) is this:

```css
<li class="guides-index guides-index-large">Switch</li>
```
This is simply nested inside the existing header: ```header #page-header``` > ```ul .nav```.

The only other tweak required is  some padding that needs to be removed for the final `li` in the header `ul` — ```body.guide header#page-header nav#feature-nav ul.nav li:last-child```. (The offset icon could also be fixed, along with the uneven spacing between the `li` elements.)

Compared to the existing design, I think a simple switch on the far side would be unobtrusive, simple, and easy to implement.

Here's a comparison of the before and after, which I think is both reasonable and functional.

![Before](https://github.com/user-attachments/assets/d7ddffca-c869-480d-ba0e-a2398b6fefb7)

![After](https://github.com/user-attachments/assets/65a44d11-9a64-4fce-b4fd-8306662dd056)

I am more than happy to make these changes to the Rails source code, but if there's any comments or advice before I do so, I would appreciate it.
